downloads/downloads-impl/src/main/java/com/duckduckgo/downloads/impl/DownloaderUtil.kt

```diff
@@ -18,21 +18,44 @@ package com.duckduckgo.downloads.impl
 
 import android.net.Uri
 import android.webkit.MimeTypeMap
-import java.lang.IllegalStateException
-import java.util.*
+import java.io.UnsupportedEncodingException
+import java.net.URLDecoder
+import java.util.Locale
 import java.util.regex.Matcher
 import java.util.regex.Pattern
+import kotlin.jvm.Throws
 import timber.log.Timber
 
 object DownloaderUtil {
 
+    private const val FILENAME_PATTERN = "\\s*(\"((?:\\\\.|[^\"\\\\])*)\"|[^;]*)\\s*"
+
+    private const val INLINE_ATTACHMENT_PATTERN = "(inline|attachment)\\s*;"
+
     // Slightly different pattern than the one used in URLUtil (also contains `inline`).
     private val CONTENT_DISPOSITION_PATTERN =
         Pattern.compile(
-            "(inline|attachment)\\s*;\\s*filename\\s*=\\s*(\"((?:\\\\.|[^\"\\\\])*)\"|[^;]*)\\s*",
+            "${INLINE_ATTACHMENT_PATTERN}\\s*filename[*]?\\s*=$FILENAME_PATTERN",
             Pattern.CASE_INSENSITIVE,
         )
 
+    private val ENCODED_FILENAME_CONTENT_DISPOSITION_PATTERN = Pattern.compile(
+        "${INLINE_ATTACHMENT_PATTERN}\\s*filename\\*\\s*=\\s*([a-z0-9\\-]+)'\\s*'$FILENAME_PATTERN",
+        Pattern.CASE_INSENSITIVE,
+    )
+
+    private val ENCODED_FILENAME_PATTERN =
```

Comment:
This can be `ENCODED_FILENAME_REGEX`. At the moment the name suggest this is a pattern, but it is a regex.